# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/xpconnect/idl/nsIXPConnect.idl
# Commit: 114c94d2677c
# Full Hash: 114c94d2677cac6b77071b302233bb93e2573fb5
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2018-12-21 16:34:09
# Regressor Bug: 1478359
# File Overlap Count: 2
# Description:
#   Bug 1480121 - Remove the global stored in nsXPCWrappedJS. r=bzbarsky
#   
#   Reasons for doing this:
#   
#   * nsXPCWrappedJS has complicated GC behavior and we're seeing some oranges in this area.
# ==============================================================================

diff -r 132871bdcd98 -r 114c94d2677c js/xpconnect/idl/nsIXPConnect.idl
--- a/js/xpconnect/idl/nsIXPConnect.idl	Thu Dec 20 21:15:11 2018 +0000
+++ b/js/xpconnect/idl/nsIXPConnect.idl	Thu Dec 20 19:13:43 2018 +0000
@@ -71,7 +71,11 @@
     readonly attribute InterfaceInfoPtr InterfaceInfo;
     readonly attribute nsIIDPtr         InterfaceIID;
 
-    // Match the GetJSObject() signature.
+    // Returns the global object for our JS object. If this object is a
+    // cross-compartment wrapper, returns the compartment's first global.
+    // The global we return is guaranteed to be same-compartment with the
+    // object.
+    // Note: this matches the GetJSObject() signature.
     [notxpcom, nostdcall] JSObjectPtr GetJSObjectGlobal();
 
     void debugDump(in short depth);